### PR TITLE
[TACHYON-1440] One worker per host configurable

### PR DESCRIPTION
**Pull request by aaudiber**
https://tachyon.atlassian.net/browse/TACHYON-1440

### CODE REVIEW

**Comment by apc999** [comment]
@kelvinchu , this PR may ease the process to run Tachyon on YARN with particular setting. Feel free to review this PR.

File: integration/bin/tachyon-yarn.sh [hunk 1]
```diff
@@ -36,5 +36,7 @@ JAR_HDFS=${PATH_HDFS}/tachyon-assemblies-${VERSION}-jar-with-dependencies.jar
 
 ${HADOOP_HOME}/bin/hadoop fs -put -f ${JAR_LOCAL} ${JAR_HDFS}
 
+export YARN_OPTS="${YARN_OPTS:-${TACHYON_JAVA_OPTS}}"
```
Comment (apc999) [inline integration/bin/tachyon-yarn.sh]: is this `YARN_OPTS` really used?

File: integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java [hunk 2]
```diff
@@ -320,9 +327,9 @@ public void stop() {
     try {
       mRMClient.unregisterApplicationMaster(FinalApplicationStatus.SUCCEEDED, "", "");
     } catch (YarnException e) {
-      LOG.error("Failed to unregister application " + e);
+      LOG.error("Failed to unregister application ", e);
```
Comment (apc999) [inline integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java]: NIT: maybe unnecessary space after "application"

File: integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java [hunk 3]
```diff
@@ -320,9 +327,9 @@ public void stop() {
...
     } catch (IOException e) {
-      LOG.error("Failed to unregister application " + e);
+      LOG.error("Failed to unregister application ", e);
```
Comment (apc999) [inline integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java]: ditto

File: integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java [hunk 4]
```diff
@@ -395,7 +402,7 @@ private void launchTachyonWorkerContainers(List<Container> containers) {
             mNMClient.startContainer(container, ctx);
             mWorkerHosts.add(container.getNodeId().getHost());
           } catch (Exception e) {
-            LOG.error("Error launching container " + container.getId() + " " + e);
+            LOG.error("Error launching container {}",container.getId(), e);
```
Comment (apc999) [inline integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java]: space before `container`?

File: libexec/tachyon-config.sh [hunk 5]
```diff
@@ -40,7 +40,7 @@ fi
 
 # A developer option to prepend Tachyon jars before TACHYON_CLASSPATH jars
 if [ -n "$TACHYON_PREPEND_TACHYON_CLASSES" ]; then
-  export CLASSPATH="$TACHYON_CONF_DIR/:$TACHYON_JARS:$TACHYON_CLASSPATH"
+  export CLASSPATH="$TACHYON_CONF_DIR/:$TACHYON_JARS:$TACHYON_CLASSPATH:$CLASSPATH"
```
Comment (apc999) [inline libexec/tachyon-config.sh]: is this fix (adding `$CLASSPATH` to the end) particularly for this PR ?

**Comment by apc999** [comment]
@aaudiber Thanks for the patch! I left some style comments.

File: integration/bin/tachyon-yarn.sh (hunk 1 above)
Comment (aaudiber) [inline integration/bin/tachyon-yarn.sh]: It determines which JVM options `tachyon.yarn.Client` is run with - I've added a comment

File: integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java (hunk 2 above)
Comment (aaudiber) [inline integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java]: Fixed, thanks

File: integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java (hunk 3 above)
Comment (aaudiber) [inline integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java]: Fixed, thanks

File: integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java (hunk 4 above)
Comment (aaudiber) [inline integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java]: Fixed, thanks

File: libexec/tachyon-config.sh (hunk 5 above)
Comment (aaudiber) [inline libexec/tachyon-config.sh]: Moved the adding to `tachyon-application-master.sh`, and added a comment on why it's needed.

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**